src/Microsoft.Data/DataFrame.cs

```diff
@@ -133,13 +133,124 @@ public DataFrame Sort(string columnName, bool ascending = true)
             for (int i = 0; i < ColumnCount; i++)
             {
                 BaseColumn oldColumn = Column(i);
-                BaseColumn newColumn = oldColumn.CloneAndAppendNulls(sortIndices, !ascending);
+                BaseColumn newColumn = oldColumn.CloneAndAppendNulls(oldColumn.NullCount, sortIndices, !ascending);
                 Debug.Assert(newColumn.NullCount == oldColumn.NullCount);
                 newColumns.Add(newColumn);
             }
             return new DataFrame(newColumns);
         }
 
+        public enum JoinAlgorithm
```

Comment:
This shouldn't be a nested enum. [https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/nested-types](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/nested-types)